README.md

```diff
@@ -0,0 +1,126 @@
+
+## 🛠️ Technologies and Tools Used
+
+This project leverages a range of modern technologies and tools to ensure robust performance and security:
+
+- **Spring Boot**: For building the backend of the application.
+- **Spring Security**: To handle authentication and authorization.
+- **Spring Data JPA**: For database operations.
+- **Swagger**: To document and test the APIs.
+- **MySQL**: As the relational database.
+- **Maven**: For project build and dependency management.
+
+## 🌟 Features
+
+### User Functionalities 🥺
+
+- **User Registration and Authentication**: Users can sign up and log in to access the store.
+- **Book Browsing and Searching**: Users can view all books, search for specific books by name, and explore books by category.
+- **Shopping Cart Management**: Users can add books to their cart, view their cart, and remove books from the cart.
+- **Order Placement and Review**: Users can place orders for books in their cart, view past orders, and see details of specific orders.
+
+### Admin Functionalities 😊
+
+- **Book Management**: Admins can add new books, update book details, and remove books from the store.
+- **Category Management**: Admins can create new categories, update category details, and delete categories.
+- **Order Management**: Admins can update the status of orders (e.g., "DELIVERED", "COMPLETED").
+
+## 🚀 Setup Instructions 🧙‍♂️🔮
+
+To set up the project locally, follow these steps:
+
+### Prerequisites
+
+Ensure you have the following software installed:
+
+- Java 21
+- Maven 3.9.5
+- MySQL
+
+### Database Configuration
+
+1. Ensure MySQL is installed and running.
+2. Create a database named `online_book_store`.
+3. Update the database configurations in `src/main/resources/application.properties`:
+
+   ```properties
+   spring.datasource.url=jdbc:mysql://localhost:3306/online_book_store
+   spring.datasource.username=your_username
+   spring.datasource.password=your_password
+   
+## Environment Variables
+Create a .env file in the root directory of your project with the following content:
+
+MYSQLDB_DATABASE=your_database_name  
+MYSQLDB_USER=your_user_name  
+MYSQLDB_ROOT_PASSWORD=your_root_password  
+MYSQLDB_LOCAL_PORT=your_local_port    
+MYSQLDB_DOCKER_PORT=your_docker_port  
+
+SPRING_LOCAL_PORT=your_local_port  
+SPRING_DOCKER_PORT=your_docker_port  
+DEBUG_PORT=your_debug_port  
+
+JWT_SECRET=your_jwt_secret  
+JWT_EXPIRATION=your_expiration_time  
+
+## Building and Running the Project ⚙️
+
+1. - **Clone the repository**:
+  ```bash
+git clone https://github.com/slizko1/Online-Book-Store.git
+cd Online-Book-Store
+  ```
+
+2. - **Build the project**:
+  ```bash
+   mvn clean package
+  ```
+
+3. **Build and start Docker containers**:
+   ```bash
+    docker-compose down --rmi all
+   ```
+   You can use this command for remove all unnecessary images
+   ```bash
+   docker-compose up --build   
+    ```
+   This command for build new images and run app in docker
```

Comment:
You should divide section 'how to install and run my app' into 2 sections: 1. No docker 2. With docker. Currently Everything is messed together